mypy/test/testpythoneval.py

```diff
@@ -80,21 +75,24 @@ def test_python_evaluation(testcase: DataDrivenTestCase) -> None:
 
     # Write the program to a file.
     program = '_' + testcase.name + '.py'
-    mypy_cmdline.append(program)
     program_path = os.path.join(test_temp_dir, program)
+    mypy_cmdline.append(program_path)
     with open(program_path, 'w') as file:
         for s in testcase.input:
             file.write('{}\n'.format(s))
+    output = []
     # Type check the program.
-    # This uses the same PYTHONPATH as the current process.
-    returncode, out = run(mypy_cmdline)
+    out, err, returncode = api.run(mypy_cmdline)
+    # split lines, remove newlines, and remove directory of test case
+    output.extend([s.rstrip('\r\n')[len(test_temp_dir + os.sep):]
```

Comment:
Hm. I don't like blindly clipping. Please use a regular for-loop that inspects the start of each line before stripping the temp directory and separator.

Reply:
Okay I will do that.